demo/lightning/checkpoint/multinode/example-deploy.yaml

```diff
@@ -0,0 +1,103 @@
+#  Copyright 2024 Google LLC
+
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+
+#       https://www.apache.org/licenses/LICENSE-2.0
+
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: my-job-run
+  labels:
+    kueue.x-k8s.io/queue-name: multislice-queue  # Name of the LocalQueue
+  annotations:
+    alpha.jobset.sigs.k8s.io/exclusive-topology: cloud.google.com/gke-nodepool # 1:1 job replica to node pool assignment
+spec:
+  failurePolicy:
+    maxRestarts: 0
+  replicatedJobs:
+    - name: checkpoint-job
+      replicas: 1
+      template:
+        spec:
+          parallelism: 10    # Equal to the number of VMs per slice
+          completions: 10    # Same as the above.
+          backoffLimit: 0   # When any pod fails, the job is failed
+          template:
+            spec:
+              schedulerName: default-scheduler
+              restartPolicy: Never
+                
+              priorityClassName: medium
+              hostNetwork: true
+              dnsPolicy: ClusterFirstWithHostNet
+              terminationGracePeriodSeconds: 30
+              containers:
+              - name: checkpoint-gpu
+                image: gcr.io/my-project/my-container
+
+                resources:
```

Comment:
Add a comment to remove this if running the CPU-based demo. Or it may even be a good idea to provide 2 example yaml files, one for CPU and one for GPU

Reply:
ACK will add CPU based demo in follow up PR (which will extend DDP strategy)